### DISCUSSION TITLE
Can you create a window with a vulkan surface and get a reference to the window?

### DISCUSSION THREAD

**Discussion by ChaseLewis** [rust-windowing/winit#2628] [Q&A] [answered]
I'm following some vulkan tutorials for rust that use winit to create the window. They use code like below to setup the surface.

```
        let surface = builder
        .build_vk_surface(&event_loop, instance.clone())
        .unwrap();
```

The issue is I don't seem to be able to get any reference to the created window when this happens. If I wanted to programmatically, minimize, restore, etc I can't do that. Is there a way to get the window reference? If not this seems like a pretty large oversight.

**Comment by kchibisov** [comment, marked as answer]
I'm not sure what you're following. Please, see the `ash` examples, since Vulkan is out of scope of winit and you have to do that with the lib you want. Ash https://github.com/ash-rs/ash
